### PR TITLE
[Serve] Add tests for Serve quickstart with ray client

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
This PR adds a test that runs the Serve "quickstart" examples from the front page of the docs using Ray Client.  

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by edoakes — APPROVED** [review]

**Comment by edoakes** [comment]
@architkulkarni test failing
https://buildkite.com/ray-project/ray-builders-pr/builds/6768#0e86b211-32ea-405d-8b2b-c498652d5c15/353-3757

File: python/ray/serve/tests/test_ray_client.py [hunk 1]
```diff
@@ -98,6 +98,58 @@ class A:
...
     assert requests.get("http://localhost:8000/A").json() == "hello"
 
+    serve.shutdown()
+    ray.util.disconnect()
+
+
+@pytest.mark.skipif(sys.platform == "win32", reason="Failing on Windows")
+def test_quickstart_class(ray_client_instance):
+    ray.util.connect(ray_client_instance, namespace="")
+    assert ray.util.client.ray.is_connected()
```
Comment (simon-mo) [inline python/ray/serve/tests/test_ray_client.py:108]: can you make these into fixture? 
```
    ray.util.connect(ray_client_instance, namespace="")
    assert ray.util.client.ray.is_connected()

    ...

    serve.shutdown()
    ray.util.disconnect()
```

Comment (architkulkarni) [reply python/ray/serve/tests/test_ray_client.py]: Good idea, done!

**Comment by architkulkarni** [comment]
Only failing test is //rllib:test_distributions which is marked flaky.

**Pull request merged**